eclipse-theia/theia
# Move plugins to user folder during resolution

#### What it does


Fixes #11463 by changing the sequence of events involved in handling `.vsix` files to place them in the user plugin folder during the resolution stage rather than the handling phase. As a consequence, 'original' source files (for example in the resolution of an 'Install from VSIX' command) are not deleted on uninstallation. 

#### How to test


1. Run the `Install from VSIX` command and select a VSIX file outside of your user plugin folders.
2. The plugin should be installed and work as expected. The original file should remain in place, and a copy should appear in the user plugins folder.
3. Uninstall the plugin in the UI.
4. You should see the 'Reload Required' message, the copy of the plugin in the user plugin folder should be deleted, and the original source should remain on disk.

---

1. Install a plugin from Open VSX.
2. A vsix file should appear in your user plugin directory, and the plugin should be available.
3. Uninstall the plugin.
4. You should see `Reload Required` in the UI, and the vsix file should be removed from your user plugin folder.
5. Click `Reload Required` to restart the front end (don't restart the backend)
6. Install the plugin again from Open VSX.
7. You should be back at step (2) with a successful installation.
> In particular, no message should be logged indicating that installation failed because the plugin was already installed at the required version.

#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
